### PR TITLE
621 bug fix baseline CVE 2021 44228

### PR DESCRIPTION
**Pull request by jonschlack**
*Added new imports and removed unused or depreciated ones
*Updated FormValue with correct values
*Removed old user guide and added new one with title, description, steps, source link, tutorial and dependencies
*Removed old LOG4j components state variables and added updated component variables
*Added useEffect function
*Added handleProcess and handleProcessTermination
*Removed killHandle function (Typo for second commit)
*Added clearOutput Docstrings
*Added RenderComponent to return statement
*Added InstallationModal to return statement
*Added LoadingOverlayAndCancelButton
*Ran npx prettier
*Cleaned up DocStrings

Exploit needs a certain version of Java runtime to execute, Oracle has removed it because it is vulnerable from download lists. 

Could not find any good or working honeypots to test the new baseline version, if you know please let me know so I can test myself.

### CODE REVIEW

File: src/components/CVE-2021-44228/CVE-2021-44228.tsx [hunk 1]
```diff
@@ -1,44 +1,56 @@
...
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+
+/**
+ * Represents the form values for the LOG4j component.
+ */
+interface FormValues {
+    attackerip: string;
+    webport: number;
+    localport: number;
```
Comment (connorwjordan) [inline src/components/CVE-2021-44228/CVE-2021-44228.tsx:18]: Should use camelCase and need to change lines these are referenced.
```suggestion
    attackerIP: string;
    webPort: number;
    localPort: number;
```

File: src/components/CVE-2021-44228/CVE-2021-44228.tsx [hunk 2]
```diff
@@ -1,44 +1,56 @@
...
+}
+// Component Constants.
+const title = "CVE-2021-44228"; // Title for the CVE 2021-44228 exploit.
 const description_userguide =
```
Comment (connorwjordan) [inline src/components/CVE-2021-44228/CVE-2021-44228.tsx:22]: Consistency with reference in checklist document and change lines this is referenced.
```suggestion
const description =
```

File: src/components/CVE-2021-44228/CVE-2021-44228.tsx (hunk 1 above)
Comment (jonschlack) [reply src/components/CVE-2021-44228/CVE-2021-44228.tsx]: Fixed issue with camelCase and description is user guide. Thank you!

**Comment by jonschlack** [comment]
Fixed reported issues and pushed a new pull request. Thank you!

File: src/components/CVE-2021-44228/CVE-2021-44228.tsx [hunk 3]
```diff
@@ -1,124 +1,243 @@
...
 
-interface FormValues {
-    attackerip: string;
-    webport: number;
-    localport: number;
-}
-
+/**
+ * @returns The LOG4j Exploit component.
+ */
```
Comment (connorwjordan) [inline src/components/CVE-2021-44228/CVE-2021-44228.tsx:42]:
```suggestion
/**
The CVE202144228 component
 * @returns The CVE202144228 component.
 */
```

File: src/components/CVE-2021-44228/CVE-2021-44228.tsx [hunk 4]
```diff
@@ -1,124 +1,243 @@
...
+import { RenderComponent } from "../UserGuide/UserGuide";
+import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+
+/**
+ * Represents the form values for the LOG4j component.
+ */
+interface FormValues {
```
Comment (connorwjordan) [inline src/components/CVE-2021-44228/CVE-2021-44228.tsx:15]:
```suggestion
interface FormValuesType {
```

File: src/components/CVE-2021-44228/CVE-2021-44228.tsx [hunk 5]
```diff
@@ -1,124 +1,243 @@
...
+            <form onSubmit={form.onSubmit(onSubmit)}>
+                <Stack>
+                    {LoadingOverlayAndCancelButton(loading, pid)}
+                    <TextInput label={"Attacker IP Address"} required {...form.getInputProps("attackerIP")} />
+                    <NumberInput label={"Web Port"} required {...form.getInputProps("webPort")} />
+                    <NumberInput label={"Local Port"} required {...form.getInputProps("localPort")} />
+                    <Button type={"submit"}>Exploit</Button>
+                    <Button type={"button"} onClick={killProcess} disabled={!pid}>
+                        Kill Process
                     </Button>
```
Comment (connorwjordan) [inline src/components/CVE-2021-44228/CVE-2021-44228.tsx:235]:
```suggestion
                    <Button type={"button"} onClick={killProcess} disabled={!pid}>Kill Process</Button>
```

**Review by connorwjordan — COMMENTED** [review]
Could not test due to no available target, and is also effected by issue #757

File: src/components/CVE-2021-44228/CVE-2021-44228.tsx (hunk 5 above)
Comment (jonschlack) [reply src/components/CVE-2021-44228/CVE-2021-44228.tsx:236]: Tried fixing but tunning prettier after adding fix but it formats the killprocess button as it was before.

File: src/components/CVE-2021-44228/CVE-2021-44228.tsx (hunk 4 above)
Comment (jonschlack) [reply src/components/CVE-2021-44228/CVE-2021-44228.tsx]: Fixed

File: src/components/CVE-2021-44228/CVE-2021-44228.tsx (hunk 3 above)
Comment (jonschlack) [reply src/components/CVE-2021-44228/CVE-2021-44228.tsx:43]: Fixed

**Comment by jonschlack** [comment]
> Could not test due to no available target, and is also effected by issue #757

Yes I know about issue, you can

> Could not test due to no available target, and is also effected by issue #757

Can you check again, a pull request has been merged that should fix the issue.

**Review by connorwjordan — APPROVED** [review]
LGTM

**Comment by michaeljpigott** [comment]
Thanks @jonschlack. Please close the corresponding Planner card now that the branch has been merged into main.

**Pull request merged**